File: app/sdks/node/lib/services/avatars.js
const Service = require('../service.js');

class Avatars extends Service {

    /**
     * Get Browser Icon
     *
     * /docs/references/avatars/get-browser.md
     *
     * @param string code
     * @param number width
     * @param number height
     * @param number quality
     * @throws Exception
     * @return {}
     */
    async getBrowser(code, width = 100, height = 100, quality = 100) {
        let path = '/avatars/browsers/{code}'.replace(new RegExp('{code}', 'g'), code);
        
        return await this.client.call('get', path, {'content-type': 'application/json'},
            {
                'width': width,
                'height': height,
                'quality': quality
            });
    }

    /**
     * Get Credit Card Icon
     *
     * /docs/references/avatars/get-credit-cards.md
     *
     * @param string code
     * @param number width
     * @param number height
     * @param number quality
     * @throws Exception
     * @return {}
     */
    async getCreditCard(code, width = 100, height = 100, quality = 100) {
        let path = '/avatars/credit-cards/{code}'.replace(new RegExp('{code}', 'g'), code);
        
        return await this.client.call('get', path, {'content-type': 'application/json'},
            {
                'width': width,
                'height': height,
                'quality': quality
            });
    }

    /**
     * Get Favicon
     *
     * /docs/references/avatars/get-favicon.md
     *
     * @param string url
     * @throws Exception
     * @return {}
     */
    async getFavicon(url) {
        let path = '/avatars/favicon';
        
        return await this.client.call('get', path, {'content-type': 'application/json'},
            {
                'url': url
            });
    }

    /**
     * Get Country Flag
     *
     * /docs/references/avatars/get-flag.md
     *
     * @param string code
     * @param number width
     * @param number height
     * @param number quality
     * @throws Exception
     * @return {}
     */
    async getFlag(code, width = 100, height = 100, quality = 100) {
        let path = '/avatars/flags/{code}'.replace(new RegExp('{code}', 'g'), code);
        
        return await this.client.call('get', path, {'content-type': 'application/json'},
            {
                'width': width,
                'height': height,
                'quality': quality
            });
    }

    /**
     * Get Image from URL
     *
     * /docs/references/avatars/get-image.md
     *
     * @param string url
     * @param number width
     * @param number height
     * @throws Exception
     * @return {}
     */
    async getImage(url, width = 400, height = 400) {
        let path = '/avatars/image';
        
        return await this.client.call('get', path, {'content-type': 'application/json'},
            {
                'url': url,
                'width': width,
                'height': height
            });
    }

    /**
     * Text to QR Generator
     *
     * /docs/references/avatars/get-qr.md
     *
     * @param string text
     * @param number size
     * @param number margin
     * @param number download
     * @throws Exception
     * @return {}
     */
    async getQR(text, size = 400, margin = 1, download = 0) {
        let path = '/avatars/qr';
        
        return await this.client.call('get', path, {'content-type': 'application/json'},
            {
                'text': text,
                'size': size,
                'margin': margin,
                'download': download
            });
    }
}

module.exports = Avatars;

File: app/sdks/dart/lib/services/avatars.dart
import "package:appwrite/service.dart";
import "package:appwrite/client.dart";
import 'package:dio/dio.dart';

class Avatars extends Service {
     
     Avatars(Client client): super(client);

     /// /docs/references/avatars/get-browser.md
    Future<Response> getBrowser({code, width = 100, height = 100, quality = 100}) async {
       String path = '/avatars/browsers/{code}'.replaceAll(RegExp('{code}'), code);

       Map<String, dynamic> params = {
         'width': width,
         'height': height,
         'quality': quality,
       };

       return await this.client.call('get', path: path, params: params);
    }
     /// /docs/references/avatars/get-credit-cards.md
    Future<Response> getCreditCard({code, width = 100, height = 100, quality = 100}) async {
       String path = '/avatars/credit-cards/{code}'.replaceAll(RegExp('{code}'), code);

       Map<String, dynamic> params = {
         'width': width,
         'height': height,
         'quality': quality,
       };

       return await this.client.call('get', path: path, params: params);
    }
     /// /docs/references/avatars/get-favicon.md
    Future<Response> getFavicon({url}) async {
       String path = '/avatars/favicon';

       Map<String, dynamic> params = {
         'url': url,
       };

       return await this.client.call('get', path: path, params: params);
    }
     /// /docs/references/avatars/get-flag.md
    Future<Response> getFlag({code, width = 100, height = 100, quality = 100}) async {
       String path = '/avatars/flags/{code}'.replaceAll(RegExp('{code}'), code);

       Map<String, dynamic> params = {
         'width': width,
         'height': height,
         'quality': quality,
       };

       return await this.client.call('get', path: path, params: params);
    }
     /// /docs/references/avatars/get-image.md
    Future<Response> getImage({url, width = 400, height = 400}) async {
       String path = '/avatars/image';

       Map<String, dynamic> params = {
         'url': url,
         'width': width,
         'height': height,
       };

       return await this.client.call('get', path: path, params: params);
    }
     /// /docs/references/avatars/get-qr.md
    Future<Response> getQR({text, size = 400, margin = 1, download = null}) async {
       String path = '/avatars/qr';

       Map<String, dynamic> params = {
         'text': text,
         'size': size,
         'margin': margin,
         'download': download,
       };

       return await this.client.call('get', path: path, params: params);
    }
}